RichEditorView/Assets/editor/rich_editor.js

```diff
@@ -480,3 +480,27 @@ RE.getRelativeCaretYPosition = function() {
     
     return y;
 };
+
+RE.getSelectionCharOffsetsWithin = function() {
+    var start = 0, end = 0;
+    var sel, range, priorRange;
+    var element = document.getElementById("main");
+    if (typeof window.getSelection != "undefined") {
+        range = window.getSelection().getRangeAt(0);
+        priorRange = range.cloneRange();
+        priorRange.selectNodeContents(element);
+        priorRange.setEnd(range.startContainer, range.startOffset);
+        start = priorRange.toString().length;
+        end = start + range.toString().length;
+    } else if (typeof document.selection != "undefined" &&
+               (sel = document.selection).type != "Control") {
+        range = sel.createRange();
+        priorRange = document.body.createTextRange();
+        priorRange.moveToElementText(element);
+        priorRange.setEndPoint("EndToStart", range);
+        start = priorRange.text.length;
+        end = start + range.text.length;
+    }
+    var string = start + ":" + end;
```

Comment:
return start + ":" + end;

Reply:
let it in a 'string' var, later it will be easier if format needs changing to suit model

Reply:
OK. But change var to let, please use ES6 syntax